src/api/procedures/removeCorporateActionsAgent.ts

```diff
@@ -30,19 +32,20 @@ export async function prepareRemoveCorporateActionsAgent(
 
   const securityToken = new SecurityToken({ ticker }, context);
 
-  const [{ owner }, agent] = await Promise.all([
-    securityToken.details(),
-    securityToken.corporateActions.getAgents(),
-  ]);
+  const agents = await securityToken.corporateActions.getAgents();
 
-  if (owner.did === agent[0].did) {
+  if (agents.length !== 1) {
     throw new PolymeshError({
       code: ErrorCode.ValidationError,
-      message: 'There is no set Corporate Actions Agent',
+      message:
+        'We can not perform this procedure with more than one Corporate Actions Agent involved',
```

Comment:
"There must be one (and only one) Corporate Actions Agent assigned to this Security Token"